server/app/controllers/admin/AdminProgramBlockPredicatesController.java

```diff
@@ -684,6 +695,28 @@ private ImmutableList<ScalarOptionElement> getScalarOptionsForQuestion(
     return scalarOptionsBuilder.build();
   }
 
+  /**
+   * Returns a list of {@link OptionElement}s representing all possible Values for the predicate,
+   * depending on question type.
+   */
+  private ImmutableList<OptionElement> getValueOptionsForQuestion(QuestionDefinition question) {
+    AtomicBoolean isFirst = new AtomicBoolean(true);
```

Comment:
(blocking) `AtomicBoolean` is usually used in multi-threaded code, this usage is rather atypical. What is the end goal trying to be accomplished?

Reply:
+1, that was the goal here.

Reply:
Alright. Seemly like a hacky use of AtomicBoolean, but we can keep it.